extensions/amp-nexxtv-player/0.1/amp-nexxtv-player.js

```diff
@@ -0,0 +1,229 @@
+    this.iframe_ = null;
+
+    /** @private {?string} */
+    this.videoIframeSrc_ = null;
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+
+    /** @private {?Promise} */
+    this.playerReadyPromise_ = null;
+
+    /** @private {?Function} */
+    this.playerReadyResolver_ = null;
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url(this.getVideoIframeSrc_(), opt_onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    this.playerReadyPromise_ = new Promise(resolve => {
+      this.playerReadyResolver_ = resolve;
+    });
+
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+
+    this.applyFillContent(iframe);
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+
+    this.element.appendChild(iframe);
+
+    installVideoManagerForDoc(this.element);
+    videoManagerForDoc(this.element).register(this);
+  }
+
+  getVideoIframeSrc_() {
+    if (this.videoIframeSrc_) {
+      return this.videoIframeSrc_;
+    }
+
+    const mediaId = user().assert(
+      this.element.getAttribute('data-mediaid'),
+      'The data-mediaid attribute is required for <amp-nexxtv-player> %s',
+      this.element);
+
+    const client = user().assert(this.element.getAttribute('data-client'),
+      'The data-client attribute is required for <amp-nexxtv-player> %s',
+      this.element);
+
+    const start = this.element.getAttribute('data-seek-to') || 0;
+    const mode = this.element.getAttribute('data-mode') || 'static';
+    const streamtype = this.element.getAttribute('data-streamtype') || 'video';
+    const origin = this.element.getAttribute('data-origin')
+      || 'https://embed.nexx.cloud/';
+
+    let src = '';
+    src += origin;
+
+    if (streamtype !== 'video') {
+      src += `${encodeURIComponent(streamtype)}/`;
+    }
+
+    src += `${encodeURIComponent(client)}/`;
+    src += `${encodeURIComponent(mediaId)}`;
+    src += `?start=${encodeURIComponent(String(start))}`;
+    src += `&datamode=${encodeURIComponent(mode)}&amp=1`;
+
+    return this.videoIframeSrc_ = src;
+  }
+
+  /** @override */
+  viewportCallback(visible) {
+    this.element.dispatchCustomEvent(VideoEvents.VISIBILITY, {visible});
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.iframe_.src = this.getVideoIframeSrc_();
+
+    this.unlistenMessage_ = listen(this.iframe_,'message', event => {
+      this.handleNexxMessages_(event);
+    });
+
+    return this.loadPromise(this.iframe_)
+      .then(() => {
+        this.element.dispatchCustomEvent(VideoEvents.LOAD);
+        this.playerReadyResolver_(this.iframe_);
+      });
+  }
+
+  pauseCallback() {
+    if (this.iframe_) {
+      this.pause();
+    }
+  }
+
+  /** @override */
+  unlayoutCallback() {
```

Comment:
We need to "unresolve" `playerReadyPromise_` be creating a new promise. We could probably just move it into `#layoutCallback`.